src/main/java/com/epam/esm/controller/GiftCertificatesController.java

```diff
@@ -58,11 +66,12 @@ public GiftCertificatesController(GiftCertificateService service) {
      *
      * @param id - id of {@link GiftCertificate} that has to be retrieved from database.
      * @return {@link ResponseEntity} contained both {@link HttpStatus} status and {@link GiftCertificateDto} object
-     * @throws EntityNotExistsException in case if nothing found with searched id.
      */
     @GetMapping(value = "/{id}", produces = JSON)
-    public ResponseEntity<GiftCertificateDto> getGiftCertificate(@PathVariable(ID) long id) throws EntityNotExistsException {
-        return new ResponseEntity<>(service.getGiftCertificate(id), HttpStatus.OK);
+    public ResponseEntity<GiftCertificateDto> getGiftCertificate(@PathVariable(ID) long id) {
+        GiftCertificateDto giftCertificate = service.getGiftCertificate(id);
+        giftCertificate.add(linkTo(methodOn(GiftCertificatesController.class).getGiftCertificate(id)).withSelfRel());
```

Comment:
You need to have more links. Add a link to all resources at least

Reply:
fixed